yearn/contracts/CurveVoterProxyStrategy.sol

```diff
@@ -0,0 +1,511 @@
+    ///         underlying tokens thus making the profit. A small portion of
+    ///         CRV rewards (defined by keepCRV param) is transferred to the
+    ///         voter contract to increase CRV boost and get more CRV rewards
+    ///         in the future. Apart from that, this strategy also sells
+    ///         the Curve pool's gauge additional rewards, generated using the
+    ///         Synthetix staking rewards contract.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        // Get the initial balance of the vault's underlying token under
+        // strategy management.
+        uint256 initialWantBalance = want.balanceOf(address(this));
+
+        // Harvest the CRV rewards from the Curve pool's gauge.
+        IStrategyProxy(strategyProxy).harvest(tbtcCurvePoolGauge);
+
+        // Buy WBTC using harvested CRV tokens.
+        uint256 crvBalance = IERC20(crv).balanceOf(address(this));
+        if (crvBalance > 0) {
+            // Deposit a portion of CRV to the voter to gain CRV boost.
+            crvBalance = adjustCRV(crvBalance);
+
+            IERC20(crv).safeApprove(dex, 0);
+            IERC20(crv).safeApprove(dex, crvBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = crv;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(dex).swapExactTokensForTokens(
+                crvBalance,
+                uint256(0),
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Claim additional reward tokens from the gauge if applicable.
+        if (tbtcCurvePoolGaugeReward != address(0)) {
+            IStrategyProxy(strategyProxy).claimRewards(
+                tbtcCurvePoolGauge,
+                tbtcCurvePoolGaugeReward
+            );
+
+            uint256 rewardBalance = IERC20(tbtcCurvePoolGaugeReward).balanceOf(
+                address(this)
+            );
+            if (rewardBalance > 0) {
+                IERC20(tbtcCurvePoolGaugeReward).safeApprove(dex, 0);
```

Comment:
Same concerns as above for the `safeApprove()` and `swapExactTokensForTokens() - deadline param`

Reply:
Same answer as above :)